.github/workflows/handle.yml

```diff
@@ -109,3 +108,85 @@ jobs:
           else
             echo "::notice title=CLI Request Handle::Skip because tool.yaml is not found"
           fi
+
+      - name: 'IAM Request Comment'
+        if: '${{ always()}}'
+        uses: 'actions/github-script@98814c53be79b1d30f795b907e553d8679345975' # ratchet:actions/github-script@v6
+        with:
+          github-token: '${{ github.token }}'
+          retries: '3'
+          script: |+
+            var req;
+            const fs = require("fs");
+            const outcome = '${{ steps.handle_iam.outcome }}';
+            switch (outcome) {
+              case 'success':
+                req = fs.readFileSync(
+                  `iam.yaml`,
+                  { encoding: "utf8" }
+                );
+
+                await github.rest.issues.createComment({
+                  owner: context.repo.owner,
+                  repo: context.repo.repo,
+                  issue_number: ${{ github.event.pull_request.number }},
+                  body: `**\`Access on Demand\`** - 🟩 **\`IAM\`** Completed request.
+
+            <details>
+            <summary>Details</summary>
+            Added below IAM permissions, and they will be expired in ${{ env.LABELED_DURATION || env.DEFAULT_DURATION }}.
+
+            \`\`\`
+            ${req}
+            \`\`\`
+            </details>
+
+            <strong> Note:</strong>
+            <strong>This is an AOD request, and merging is NOT allowed.</strong> The request will be automatically applied once the code owners approve it. Please close the PR once you are finished.
+            For more instructions, please see [here](https://github.com/abcxyz/aod-template/blob/main/README.md).`,
+                });
+                break;
+              case 'failure':
+                req = fs.readFileSync(
+                  `iam.yaml`,
+                  { encoding: "utf8" }
+                );
+                const error = fs.readFileSync(
+                  `${{ env.IAM_ERROR_FILENAME }}`,
+                  { encoding: "utf8" }
+                );
+                await github.rest.issues.createComment({
+                  owner: context.repo.owner,
+                  repo: context.repo.repo,
+                  issue_number: ${{ github.event.pull_request.number }},
+                  body: `**\`Access on Demand\`** - 🟥 **\`IAM\`** Failed to completed request.
+
+            <details>
+            <summary>Details</summary>
+            Failed to added below IAM permissions.
+
+            \`\`\`
+            ${req}
+            \`\`\`
+
+            Error:
+            \`\`\`
+            ${error}
+            \`\`\`
+            </details>
+
+            <strong> Note:</strong>
+            <strong>This is an AOD request, and merging is NOT allowed.</strong> The request will be automatically applied once the code owners approve it. Please close the PR once you are finished.
+            For more instructions, please see [here](https://github.com/abcxyz/aod-template/blob/main/README.md).`,
```

Comment:
Can we have this comment only once for a PR? Probably on PR open event.

Reply:
What if a later commit delete the file and do sth else?